Task U5. Cookies

ShulyakSA opened the pull request:
- Added methods for writing cookies to a file and reading cookies from it in MainSteps
- Added parametrized test that login on the first run and uses cookies on the second
- Added profiles in pom.xml for parameterize the path to the file with suites testng.xml, testngSqlEx.xml, testngW2A.xml
- Some changes in test classes and configs for support loading pages from different sites

ShulyakSA committed:
Task U5. Cookies
- Added methods for writing cookies to a file and reading cookies from it in MainSteps
- Added parametrized test that login on the first run and uses cookies on the second
- Added profiles in pom.xml for parameterize the path to the file with suites testng.xml, testngSqlEx.xml, testngW2A.xml
- Some changes in test classes and configs for support loading pages from different sites

sun6r0 commented on src/test/java/steps/MainSteps.java:
```diff
@@ -46,4 +53,65 @@ public static String getText(WebDriver driver, WebElement element) {
+            file.createNewFile();
+            FileWriter fileWrite = new FileWriter(file);
+            BufferedWriter bWrite = new BufferedWriter(fileWrite);
+            for (Cookie cookie : driver.manage().getCookies()) {
+                bWrite.write((cookie.getName() + "|" + cookie.getValue() + "|" + cookie.getDomain() + "|" + cookie.getPath() + "|" + cookie.getExpiry() + "|" + cookie.isSecure()));
+                bWrite.newLine();
```
Все параметры необязательно было сохранять для авторизации, можно было бы сделать так  fileWrite.write(driver.manage().getCookieNamed(cookieName).getValue());

ShulyakSA replied:
Переписал

sun6r0 commented on src/test/java/tests/AuthorizationWithCookiesTest.java:
```diff
@@ -0,0 +1,33 @@
+    @Test(description = "Проверка авторизации c Cookies")
+    @Parameters("invocationCount")
+    public void authorizationWithCookiesTest(int invocationCount) {
+        if (invocationCount == 1) {
+            sqlExPage.basicAuth();
+        } else sqlExPage.authorizationWithCookies();
```
Вместо конструкции с if лучше логинься, сохраняй куку, удаляй драйвером все куки и логинься снова.  Конструкции с условиями не очень приветствуются в тестах обычно

ShulyakSA replied:
Убрал параметризацию и условие

ShulyakSA committed:
Task U5. Cookies
- Added methods for writing cookies to a file and reading cookies from it in MainSteps
- Added parametrized test that login on the first run and uses cookies on the second
- Added profiles in pom.xml for parameterize the path to the file with suites testng.xml, testngSqlEx.xml, testngW2A.xml
- Some changes in test classes and configs for support loading pages from different sites
